pkg/controllers/clusterexternalsecret/clusterexternalsecret_controller_test.go

```diff
@@ -291,12 +312,19 @@ var _ = Describe("ClusterExternalSecret controller", func() {
 		}
 		tc.checkClusterExternalSecret = func(ces *esv1beta1.ClusterExternalSecret) {
 			for _, namespace := range tc.externalSecretNamespaces {
-				var es esv1beta1.ExternalSecret
-				err := k8sClient.Get(context.Background(), types.NamespacedName{
-					Namespace: namespace.namespace.Name,
-					Name:      ExternalSecretName,
-				}, &es)
-				Expect(err).ToNot(HaveOccurred())
+				Eventually(func() bool {
```

Comment:
This is not directly related to the PR but this test seems to be very flaky since it assumes external secrets exist right after the CES is created. However, it is not always the case, so we should use `Eventually` for asserting their existence.